src/tests/desktop-agent.spec.js

```diff
@@ -54,18 +83,56 @@ describe("Desktop Agent", function() {
       it("is defined", function() {
         expect(desktopAgent.findIntent).toBeDefined();
       });
+      it("returns an AppIntent object", function() {
+        return desktopAgent.findIntent("StartChat").then(function(result) {
+          expect(result).toBeDefined();
+          expect(result.intent).toBeDefined();
+          expect(result.apps).toBeDefined();
+        });
+      });
     });
 
     describe("findIntentsByContext()", function() {
       it("is defined", function() {
         expect(desktopAgent.findIntentsByContext).toBeDefined();
       });
+      it("returns an array of AppIntent objects", function() {
+        let context = {};
+        return desktopAgent.findIntentsByContext(context).then(function(result) {
+          expect(result).toBeDefined();
+          result.forEach(val => {
+            expect(val.intent).toBeDefined();
+            expect(val.apps).toBeDefined();
+          });
+        });
+      });
     });
 
     describe("raiseIntent()", function() {
       it("is defined", function() {
         expect(desktopAgent.raiseIntent).toBeDefined();
       });
+
+      it("returns an IntentResolution object", function() {
+        let context = {};
+        return desktopAgent.raiseIntent("StartChat", context).then(function(result) {
+          expect(result).toBeDefined();
+          expect(result.source).toBeDefined();
+          expect(result.version).toBeDefined();
+        });
+      });
+
+      it("returns an error if app was not found", function() {
+        let error = desktopAgent.raiseIntent();
+        return error.then(val => {}, reason => {
+          let expected = {
+            source: null,
+            version: ''
+          };
+          expect(reason).toEqual(expected);
```

Comment:
Would the test have access to the `reason` function here since the `expect` is inside of the return value of the `reason` function?

Reply:
I'm not sure I understand the question. The test runs and I'm able to access variables in the `reason` scope. Since `error` is a rejected promise, calling `then()` will trip the `reason` function with `reason` as the parameter.

Reply:
You're right, completely misread that arrow function last night 😅